File: Classes/Attacks/BossAttack.py
from random import randint
import parameters as param
from Classes.Attacks.Attacks import Attack

class AtkChefe(Attack):
    def __init__(self, nomeAtk, danoAtk, spriteSheet, limiteLoops,
                 qtdFrames, pixels, spritePosition, offsetFrames, hitKillChance):
        """
        :param nomeAtk: str
        :param danoAtk: int
        :param spriteSheet: spriteSheet loaded by pygame.image.load()
        :param limiteLoops: int
        :param pixels: (int, int) (pixels width, pixels height)
        :param qtdFrames: int
        :param spritePosition: (float, float)
        :param offsetFrames: float (dictates the animation speed)
        :param hitKillChance: int
        """

        Attack.__init__(self, nomeAtk, danoAtk, spriteSheet, limiteLoops,
                        qtdFrames, pixels, spritePosition, offsetFrames)

        self._hitKillChance = hitKillChance

    def randomizeHitKill(self):
        randomizer = randint(1, 100)
        if randomizer <= self._hitKillChance:
            self.setDamage(param.playerHp)

    def getHitKillChance(self):
        return self._hitKillChance

    def setHitKillChance(self, c):
        self._hitKillChance = c

File: TheFlameRealmJava/core/src/main/java/com/flamerealm/game/attacks/BossAttack.java
package com.flamerealm.game.attacks;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.math.GridPoint2;
import com.badlogic.gdx.math.MathUtils;
import com.badlogic.gdx.math.Vector2;
import com.flamerealm.game.GameConstants;

/**
 * Port de BossAttack.py (AtkChefe).
 */
public class BossAttack extends Attack {
    private int hitKillChance;

    public BossAttack(String name, int damage, Texture spriteSheet, int limiteLoops,
                       int qtdFrames, GridPoint2 pixels, Vector2 spritePosition, float offsetFrames,
                       int hitKillChance) {
        super(name, damage, spriteSheet, limiteLoops, qtdFrames, pixels, spritePosition, offsetFrames);
        this.hitKillChance = hitKillChance;
    }

    public void randomizeHitKill() {
        int randomizer = MathUtils.random(1, 100);
        if (randomizer <= hitKillChance) {
            setDamage(GameConstants.playerHp);
        }
    }

    public int getHitKillChance() {
        return hitKillChance;
    }

    public void setHitKillChance(int hitKillChance) {
        this.hitKillChance = hitKillChance;
    }
}
